sound/soc/sof/intel/hda-codec.c

```diff
@@ -124,10 +123,16 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 	if (hda_codec_use_common_hdmi ||
 	    (resp & 0xFFFF0000) != IDISP_VID_INTEL) {
 		hdev->type = HDA_DEV_LEGACY;
-		hda_codec_load_module(&hda_priv->codec);
+		ret = hda_codec_load_module(&hda_priv->codec);
+		/*
+		 * handle ret==0 (no driver bound) as an error, but pass
+		 * other return codes without modification
```

Comment:
@kv2019i wht is the return value when device_attach succeeds?

Reply:
@ranj063 
From dd.c docs for device_attach():
* device_attach - try to attach device to a driver.                                                                                                                               
[...]
 * Returns 1 if the device was bound to a driver;                                                                                                                                  
 * 0 if no matching driver was found;                                                                                                                                              
 * -ENODEV if the device is not registered.   

So it returns 1 if device is bound (already before, or as a result of the attach call). 0 means device exist but no matching driver. And then various negative error codes.

Reply:
Thanks! This looks good now